pkg/management/component.go

```diff
@@ -585,17 +585,25 @@ func (c *component) UpdateUser(ctx context.Context, realmName, userID string, us
 	}
 
 	// when the email changes, set the EmailVerified to false
-	if user.Email.Defined && (user.Email.Value == nil || c.isUpdated(user.Email.Value, oldUserKc.Email)) {
-		var verified = false
-		user.EmailVerified = &verified
+	if user.Email.Defined {
+		if user.Email.Value == nil {
+			var verified = false
+			user.EmailVerified = &verified
+		} else if c.isUpdated(user.Email.Value, oldUserKc.Email) {
+			if err = c.checkEmailNotUsed(ctx, realmName, userID, *user.Email.Value); err != nil {
+				c.logger.Info(ctx, "msg", "Can't update email with email of an already existing account", "email", *user.Email.Value)
+				return err
+			}
+			oldUserKc.SetAttributeString(constants.AttrbEmailToValidate, *user.Email.Value)
+		}
 	}
 
 	// when the phone number changes, set the PhoneNumberVerified to false
-	if user.PhoneNumber.Defined && (user.PhoneNumber.Value == nil || c.isUpdated(user.PhoneNumber.Value, oldUserKc.GetAttributeString(constants.AttrbPhoneNumber))) {
-		var verified = false
-		user.PhoneNumberVerified = &verified
+	if user.PhoneNumber.Defined {
 		if user.PhoneNumber.Value == nil {
 			removeAttributes = append(removeAttributes, constants.AttrbPhoneNumber, constants.AttrbPhoneNumberVerified)
```

Comment:
Should we also remove phoneNumberToValidate ?

Reply:
Good point... do we want to remove validated phone number or just the not-yet-validated one?

Reply:
Repushed as discussed together